cmd/bucket-metadata-sys.go

```diff
@@ -525,38 +516,49 @@ func (sys *BucketMetadataSys) refreshBucketsMetadataLoop(ctx context.Context) {
 			}
 			sys.RemoveStaleBuckets(diskBuckets)
 
-			for _, bucket := range buckets {
+			for i := range buckets {
 				wait := sleeper.Timer(ctx)
 
-				err := sys.loadBucketMetadata(ctx, bucket)
+				meta, err := loadBucketMetadata(ctx, sys.objAPI, buckets[i].Name)
 				if err != nil {
 					logger.LogIf(ctx, err)
 					wait() // wait to proceed to next entry.
 					continue
 				}
 
+				sys.Lock()
+				sys.metadataMap[buckets[i].Name] = meta
+				sys.Unlock()
+
+				// Initialize the failed buckets
+				if _, ok := failedBuckets[buckets[i].Name]; ok {
+					globalEventNotifier.set(buckets[i], meta)
```

Comment:
If you ask me, we should also call globalEventNotifer.set() and globalBucktTargetSys.set(); because let's say one node did not receive an update from a peer about this configuration change; this will still be not updated. 

I think we need to add a timestamp in all configurations so we know when we update and when not; but not in this PR; in another one instead.

Reply:
@Praveenrajmani will merge this PR please send an another one.

Reply:
These `set` calls doesn't do any peer rest calls. It is just a local in-memory update.